### PR TITLE
Fixes #9012: Remove migration scripts for 2.10-3.0 in 4.0

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9012

### CODE REVIEW

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 1]
```diff
@@ -341,9 +328,6 @@ upgrade_ncf() {
   fi
 
   if [ -f ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf ]; then
-    # 2.9.0 & 2.11.11, 3.0.5: Make sure that ncf uses the right logger bundles (_logger_default and logger_rudder)
-    STEP="Make sure that ncf uses the right logger bundles (_logger_default and logger_rudder)"
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade]: I think the comment should stay, but in a modified form, reflecting what is done below:
Make sure that ncf uses the right logger bundles (_log_default and log_rudder)"

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 2]
```diff
@@ -359,14 +343,6 @@ upgrade_ncf() {
       sed -i "s%^loggers=\(.*\)%loggers=\1,log_rudder%" ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf
     fi
 
-    # 2.9.0: Make sure that ncf uses the right CFEngine port (5309)
-    STEP="Make sure that ncf uses the right CFEngine port (5309)"
-
-    if ! cat ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf | perl -p0e "s/\n/%%%%%/g" | egrep -q "classes.community_edition[^%]*%%%%%cfengine_port=5309"; then
-      cat ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf | perl -p0e "s/\n/%%%%%/g" | sed "s/\(classes.community_edition[^%]*%%%%%cfengine_port=\)[0-9]\+/\15309/" | perl -p0e "s/%%%%%/\n/g" > ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf.new
-      mv ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf.new ${CONFIGURATION_REPOSITORY}/ncf/ncf.conf
-    fi
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade]: oh, nice catch. it took me a while to understand that it was indeed not needed (ncf.conf has a lot of magic)

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 3]
```diff
@@ -670,21 +557,6 @@ upgrade_inventory() {
   # Check if the LDAP has been initialised
   LDAP_EXISTS=$(${LDAPSEARCH} -b "cn=rudder-configuration" -s base dn 2>/dev/null | grep -c "dn: cn=rudder-configuration" || true)
 
-  ## 2.4.0 : Change attribute from dn:ruleId=inventory-all,ou=Rules,ou=Rudder,cn=rudder-configuration
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade]: it was about time to remove it :)

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 4]
```diff
@@ -44,23 +44,10 @@ trap anomaly_handler ERR INT TERM
...
-# - 2.11.6       : Remove LDAP indexes
-# - 2.11.6       : Add LDAP cache optimizations
 # - 2.11.19      : Add index on fileformat for eventlog
 # - 2.11.23      : Add 'api compatibility' property
-# - 3.0.0        : Add the nodeConfigId column in the ReportsExecution table and add the nodes_info table
-# - 3.0.0        : Add the rudder.jdbc.maxPoolSize property
-# - 3.0.0        : File Format migration from 5 to 6
-# - 3.0.0        : Add the insertionId column in the ReportsExecution table and fill its value for last nodes run
-# - 3.0.0        : Replace varchar by text in the database
-# - 3.0.2        : Check for the PostgreSQL version (>= 8.4)
```
Comment (ncharles) [inline rudder-webapp/SOURCES/rudder-upgrade]: i didn't see that one removed below, are you sure you want to remove it ?

**Comment by peckpeck** [comment]
Commit modified

**Comment by fanf** [comment]
@ncharles: could you review the modification and merge if OK?

File: rudder-webapp/SOURCES/rudder-upgrade [hunk 5]
```diff
@@ -550,14 +471,7 @@ rudder.server-roles.cfengine-mission-portal=autodetect
...
-  # - 3.0.0 : File Format migration from 5 to 6
-  STEP="3.0.0: File Format migration from 5 to 6"
-
-  RES=$(${PSQL} -t -d rudder -c "select count(*) from (select ((xpath('/entry//@fileFormat',data))[1]::text) as id from eventlog) as T where id='5';")
-  if [ $RES -ne 0 ]; then
-    ${PSQL} -d rudder -f ${RUDDER_UPGRADE_TOOLS}/dbMigration-2.11-3.0-set-migration-needed-flag-for-EventLog.sql > /dev/null 2>&1
-  fi
-
+  :
 }
```
Comment (jooooooon) [inline rudder-webapp/SOURCES/rudder-upgrade]: Why not remove the function?

**Pull request merged**